src/pages/hacks/re-entrancy/ReEntrancy.sol

```diff
@@ -58,16 +58,16 @@ contract Attack {
         etherStore = EtherStore(_etherStoreAddress);
     }
 
-    // Fallback is called when EtherStore sends Ether to this contract.
-    fallback() external payable {
-        if (address(etherStore).balance >= 1 ether) {
+    // Receive is called when EtherStore sends Ether to this contract
+    receive() external payable {
+        if (address(etherStore).balance >= msg.value) {
```

Comment:
This check is incorrect.

For example, if EthStore has 0.5 ETH and attacker deposits 1 ETH. The re-entrancy attack will fail.
First withdraw will send 1 ETH, second withdraw will try to send 1 ETH again, but there is only 0.5 in `EthStore`.

Hence the amount needs to be hard-coded to match the amount deposited by the attacker. Store this either in a constant or a state variable

Reply:
That's correct! I hardcoded a constant then, looks cleaner to look at